main.go

```diff
@@ -167,3 +171,38 @@ func bubbleSort(input [][]byte, comp func(a, b []byte) int) [][]byte {
 
 	return input
 }
+
+func mergeSort(input [][]byte, comp func(a, b []byte) int) [][]byte {
+	if len(input) < 2 {
+		return input
+	}
+
+	mid := len(input) / 2
+
+	mergeDuplicate := make([][]byte, len(input))
+	for i := 0; i < len(input); i++ {
+		mergeDuplicate[i] = make([]byte, len(input))
+		for j := 0; j < len(input[0]); j++ {
+			mergeDuplicate[i][j] = input[i][j]
+		}
+	}
+
+	for i := 0; i < len(input); i++ {
+		if comp(input[i],input[i+1]) > 0 {
+			mergeDuplicate[i], mergeDuplicate[i+1] = input[i+1], input[i]
+			return input
+		}
+	}
+
+	fmt.Println(mergeSort(input[:mid], comp))
+	fmt.Println(mergeSort(input[mid:], comp))
```

Comment:
You have to store mergeSort()s results into variables **and then** merge them into mergeDuplicate.